### PR TITLE
Restrict several calendar events of one type with intersected dates

### PR DESCRIPTION
**Pull request by sponkrashin**
Issue #340

### CODE REVIEW

**Comment by sponkrashin** [comment]
We should, because if anybody calls API request to add vacation with same dates - it will break CSP database.

File: server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs [hunk 1]
```diff
@@ -20,10 +20,10 @@ public sealed class DatesPeriod
 
         public DatesPeriod(DateTime startDate, DateTime endDate, int startWorkingHour = 0, int finishWorkingHour = 8)
         {
-            this.StartDate = startDate;
-            this.EndDate = endDate;
-            this.StartWorkingHour = startWorkingHour;
-            this.FinishWorkingHour = finishWorkingHour;
+            this.StartDate = MinDate(startDate, endDate);
+            this.EndDate = MaxDate(startDate, endDate);
+            this.StartWorkingHour = Math.Min(startWorkingHour, finishWorkingHour);
```
Comment (santee) [inline server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: lol

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: It is just to be sure that all first/start params are normalized

Comment (santee) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: I understand that :)

Comment (santee) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: but better to throw an error when created? It's invalid data, after all

Comment (santee) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: Somewhere in controller

File: server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs [hunk 2]
```diff
@@ -63,5 +63,30 @@ public override int GetHashCode()
...
         }
+
+        public bool DatesIntersectsWith(DatesPeriod period)
+        {
+            if (period == null)
+            {
+                return false;
+            }
+
+            if (this.EndDate < period.StartDate || period.EndDate < this.StartDate)
```
[the remark this answers is not in the document]
Comment (santee) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: nvm, too complicated

File: server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs (hunk 1 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: It is checked really, it doesn't allow to create event when Finish is more than Start.

Comment (santee) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: what's the purpose of this code then? :) Anyway, better save than sorry?

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar.Abstractions/DatesPeriod.cs]: Purpose is to not rely on server handles this case :)

**Pull request merged**